src/cmd/extrinsics/call.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+// This file is part of cargo-contract.
+//
+// cargo-contract is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// cargo-contract is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::Transcoder;
+use crate::ExtrinsicOpts;
+use anyhow::Result;
+use jsonrpsee::common::Params;
+use serde::{Deserialize, Serialize};
+use sp_core::Bytes;
+use sp_rpc::number::NumberOrHex;
+use std::convert::TryInto;
+use structopt::StructOpt;
+use subxt::{
+    balances::Balances, contracts::*, system::System, ClientBuilder, ContractsTemplateRuntime,
+    ExtrinsicSuccess, Signer,
+};
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "call", about = "Call a contract")]
+pub struct CallCommand {
+    /// The name of the contract message to call
+    name: String,
+    /// The call arguments, encoded as strings
+    args: Vec<String>,
+    #[structopt(flatten)]
+    extrinsic_opts: ExtrinsicOpts,
+    /// Maximum amount of gas to be used for this command
+    #[structopt(name = "gas", long, default_value = "50000000000")]
+    gas_limit: u64,
+    /// The value to be transferred as part of the call
+    #[structopt(name = "value", long, default_value = "0")]
+    value: <ContractsTemplateRuntime as Balances>::Balance,
+    #[structopt(name = "contract", long)]
```

Comment:
```suggestion
    #[structopt(name = "contract", long, env = "CONTRACT")]
```

This will use the environment variable `CONTRACT` as a fallback if a) it exists and  b) `--contract` was not given on the command line – thus enabling kind of a "session mode" :-).

Reply:
TIL :rocket: